### DISCUSSION TITLE
RTC methods missing?

### DISCUSSION THREAD

**Discussion by smhodge42** [micropython/micropython#13578] [ESP32]
I am trying to use the machine.RTC class on an Espressif ESP32-S3-DevKitC board but it seems like the documentation (yes, the latest) describes methods that do not exist.  The firmware is the latest available, 1.22.1. Here is a REPL session, using rshell:

Entering REPL. Use Control-X to exit.
>
MicroPython v1.22.1 on 2024-01-05; Generic ESP32S3 module with ESP32S3
Type "help()" for more information.
>>>
>>> from machine import RTC
>>> rtc=RTC()
>>> rtc.datetime()
(2024, 1, 31, 2, 13, 41, 37, 726144)
>>> rtc.now()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'RTC' object has no attribute 'now'
>>>

The same is true of the alarm() and irq() methods (and maybe others, I did not test all of them, only the ones I need).

**Comment by robert-hh** [comment]
Not all ports implement all methods. There is a note in the documentation. About RTC, see issues #10578 and #5553 and PR #10607.
RTC.now() is redundant. You can as well use RTC.datetime().

**Reply by smhodge42** [reply]
Thanks. I can easily subsitute datetime() for now() but my real need is to have an ISR triggered periodically by the RTC. For a pyboard I use wakup(), no problems. For the ESP32 the documentation indicates use alarm() instead, but that too does not seem to exist in the ESP23 port (latest version), and the same for the corresponding irq() method. So is it not possible to trigger a RTC ISR on an ESP32 port even though the ESP32 documentation implies it can be done? If not it is huge obstacle for me to migrate code from pyboard to ESP32.

**Reply by ned-pcs** [reply]
Could you use `machine.deepsleep()` to sleep for a given number of milliseconds instead?

**Reply by smhodge42** [reply]
No, I've got other stuff I need to do, all unpredictable interupt-driven. I have a crude work-around using a timer to periodically read the RTC every 250 ms, but that's polling and by no means ideal.

**Reply by ricksorensen** [reply]
You can use the Timer class (https://docs.micropython.org/en/latest/library/machine.Timer.html) to trigger periodic interrupts?  Will this suffice for your needs?  I verified this works as a periodic interrupt on as C3 and S3 board today.
